src/stan-users-guide/dae.Rmd

```diff
@@ -0,0 +1,213 @@
+explicitly depends on `t`, however one still needs to specify `t` as
+an argument.
+
+### Strict signature {-}
+
+The types in the DAE residual function are strict. The first argument is the time
+passed as a `real`, the second argument is the state passed as a `vector`,
+the third argument is the state derivative passed as a `vector`,
+and the return type is a `vector`. A model that does not have this signature will
+fail to compile. The fourth argument onwards can be any type, granted all
+the argument types match the types of the respective arguments in the solver
+call.
+
+All of these are possible DAE signatures:
+
+```stan
+vector my_dae1(real t, vector y, vector yp, real a0);
+vector my_dae2(real t, vector y, vector yp, array[] int a0, vector a1);
+vector my_dae3(real t, vector y, vector yp, matrix a0, array[] real a1, row_vector a2);
+```
+
+but these are not allowed:
+
+```stan
+vector my_dae1(real t, array[] real y, vector yp); 
+// Second argument is not a vector
+array[] real my_dae2(real t, vector y, vector yp); 
+// Return type is not a vector
+vector my_dae3(real t, vector y); 
+// First argument is not a real and missing the third argument
+```
+
+## Solving DAEs
+
+Stan provides a `dae` function for solving DAEs, so that the above chemical reaction
+equation can be solved in the following code.
+```stan
+data {
+  int N;
+  vector[3] yy0;
+  vector[3] yp0;
+  real t0;
+  real alpha;
+  real beta;
+  array[N] real ts;
+  array[N] vector[3] y;
+}
+parameters {
+  real gamma;
+}
+transformed parameters {
+  vector[3] y_hat[N] = dae(chem, yy0, yp0, t0, ts, alpha, beta, gamma);
+}
+```
+<a name="id:dae-solve.figure"></a>
+
+Since `gamma` is a parameter, the DAE solver is called in the transformed parameters block. 
+
+## Control parameters for DAE solving {#control-ode.section}
+
+Using `dae_tol` one can specify the `relative_tolerance`, `absolute_tolerance`, and
+`max_num_steps` parameters in order to control the DAE solution.
+
+```stan
+vector[3] y_hat[N] = dae_tol(chem, yy0, yp0, t0, ts,
+                             relative_tolerance,
+                             absolute_tolerance,
+                             max_num_steps,
+                             alpha, beta, gamma);
+```
+
+`relative_tolerance` and `absolute_tolerance` control accuracy the solver tries to achieve, and
+`max_num_steps` specifies the maximum number of steps the solver will
+take between output time points before throwing an error.
+
+The control parameters must be data variables -- they cannot be
+parameters or expressions that depend on parameters, including local
+variables in any block other than transformed data and generated
+quantities. User-defined function arguments may be qualified as only
+allowing data arguments using the `data` qualifier.
+
+The default value of relative and absolute
+tolerances are $10^{-10}$ and the maximum number of steps between outputs is
+one hundred million.
```

Comment:
These defaults seem to be conservative and if they are we could mention it. On the other hand, what is or isn't conservative probably depends on the problem at hand.

Reply:
added
> We suggest the user choose the control parameters according
to the problem in hand, and resort to the defaults only when no
knowledge of the DAE system or the physics it models is available.